tests/curve_fit_test.py

```diff
@@ -163,11 +155,6 @@ def test_compare_to_curve_fit_xarray(dims, coords, reduce_dims, p0, params, boun
             result[param_name].data,
             xresult['param', sc.scalar(param_name)].data,
         )
-        if (
-            bounds is None
-            or bounds[param_name][0] <= params[param_name] <= bounds[param_name][1]
-        ):
-            assert sc.allclose(result[param_name].data, sc.scalar(params[param_name]))
```

Comment:
Why did you remove this now?

Reply:
Because after replacing `func` and `func2d` with `func3d` in the array creation the result of the fit will depend on `t` and `y` in a way that I think is difficult to predict. Therefore I only kept the comparison with xarray. We test the accuracy of the results in other tests anyway.